core/src/main/java/org/apache/accumulo/core/client/mapred/AbstractInputFormat.java

```diff
@@ -384,7 +387,21 @@ public static InputTableConfig getInputTableConfig(JobConf job, String tableName
   protected abstract static class AbstractRecordReader<K,V> implements RecordReader<K,V> {
     protected long numKeysRead;
     protected Iterator<Map.Entry<Key,Value>> scannerIterator;
-    protected RangeInputSplit split;
+    protected org.apache.accumulo.core.client.mapreduce.impl.AccumuloInputSplit split;
+    protected ScannerBase scannerBase;
+
+    /**
+     * Configures the iterators on a scanner for the given table name.
+     *
+     * @param job
+     *          the Hadoop job configuration
+     * @param scanner
+     *          the scanner for which to configure the iterators
+     * @param tableName
+     *          the table name for which the scanner is configured
+     * @since 1.7.0
+     */
+    protected abstract void setupIterators(JobConf job, ScannerBase scanner, String tableName, AccumuloInputSplit split);
```

Comment:
I like how your suggestion moves configuration of the scanner to a private method.  Trying to understand your suggestion made me realize I don't understand why `AccumuloInputSplit` is storing and serializing things like iterators.    Why not get that info from the job conf when the task runs?  I am being lazy asking this question, let me know if its not quick/easy to explain I can dig around in the code.